File: dotnet/Multiplayer/Rooms/RoomPlayersResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Rooms
{
    /// <summary>
    /// Response containing the list of players in a game room.
    /// </summary>
    /// <typeparam name="T">The type to deserialize player data into.</typeparam>
    public class RoomPlayersResponse<T> : ApiResponse<ERoomPlayersError> where T : class, new()
    {
        /// <summary>
        /// List of players currently in the game room.
        /// </summary>
        public List<RoomPlayer<T>> Players { get; set; } = new();

        /// <summary>
        /// Timestamp of when the player list was last updated.
        /// </summary>
        public string Last_updated { get; set; } = string.Empty;
    }
}

File: java/src/main/java/com/michitai/multiplayer/rooms/RoomPlayersResponse.java
package com.michitai.multiplayer.rooms;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

import java.util.List;

/**
 * Response containing the list of players in a game room.
 *
 * @param <T> The type to deserialize player data into.
 */
public class RoomPlayersResponse<T> extends ApiResponse {
    @JsonProperty("players")
    private List<RoomPlayer<T>> players;

    @JsonProperty("last_updated")
    private String lastUpdated;

    public List<RoomPlayer<T>> getPlayers() {
        return players;
    }

    public void setPlayers(List<RoomPlayer<T>> players) {
        this.players = players;
    }

    public String getLastUpdated() {
        return lastUpdated;
    }

    public void setLastUpdated(String lastUpdated) {
        this.lastUpdated = lastUpdated;
    }
}
